.github/workflows/scanner-build.yaml

```diff
@@ -237,11 +282,67 @@ jobs:
 
         push_scanner_image_manifest_lists "$architectures"
 
+  scan-images-with-roxctl:
+    if: github.event_name == 'push'
+    needs:
+    - build-and-push-scanner
+    - push-scanner-manifests
+    name: Check images for vulnerabilities
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed for stackrox/central-login to create the JWT token.
+      id-token: write
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          [
+            "scanner-v4",
+            "scanner-v4-db",
+          ]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - uses: ./.github/actions/job-preamble
+      with:
+        gcp-account: ${{ secrets.GCP_SERVICE_ACCOUNT_STACKROX_CI }}
+
+    - uses: ./.github/actions/handle-tagged-build
+
+    - name: Central login
+      uses: stackrox/central-login@v1
+      with:
+        endpoint: ${{ vars.ACS_DOGFOODING_CENTRAL_URL }}
```

Comment:
Since we're logging into `STACKROX_CI_INSTANCE_CENTRAL`, should we skip this job on PRs from forks? In the same way we skip certain steps in `build.yaml`, e.g., `Docker login`:
https://github.com/stackrox/stackrox/blob/dc2c4f21825eba5b76e7de529b6233b43fc63817/.github/workflows/build.yaml#L380-L383

If so, we probably want to do the same when scanning the rest of the images here: https://github.com/stackrox/stackrox/blob/master/.github/workflows/build.yaml#L644

cc @tommartensen @janisz @gavin-stackrox

Reply:
We only run this step upon push to the `master` branch (or the creation of a `nightly` if I'm reading the `on` conditions correctly at the stop of the file), so I think it's ok as-is. I don't believe this will run on PRs, fork or not.

Please do correct me if I'm wrong. I'm not a GitHub workflow expert at all, so I could be misreading this

Reply:
It will also run on PRs: https://github.com/stackrox/stackrox/pull/11443/files#diff-9ff08f51e9e9b767c00bc7066e8be6337cc5661d029248adc9557a9e6ab5fa3eL10-L14 - when a PR is opened, pushed to or reopened. 

Therefore, it'd be best if the external contribution conditions were applied.

Reply:
This specific step has the `push` condition, though: https://github.com/stackrox/stackrox/pull/11443/files#diff-9ff08f51e9e9b767c00bc7066e8be6337cc5661d029248adc9557a9e6ab5fa3eR286

Reply:
It was skipped in this PR: https://github.com/stackrox/stackrox/actions/runs/9508087940/job/26209112458?pr=11443

Reply:
You're right. Since you have the same condition, it should be fine regarding secrets.